### DISCUSSION TITLE
OSX homebrew installation of libvoikko

### DISCUSSION THREAD

**Discussion by jarirajari** [voikko/discussion#4] [General]
Hi,

I am facing some problems with libvoikko. I am using MacOS Ventura (version 13.2) - arm64 arch. So basically I installed "libvoikko" like this "brew install libvoikko". On the Python3.9 side, "import libvoikko" works, but initializing it e.g. with "voikko = libvoikko(u"fi")" fails due to a library missing. The lib in question is "libvoikko.1.dylib", that was searched from '/System/Volusmes/Preboot/Cryptexes/OSlibvoikko.1.dylib', '/usr/lib/libvoikko.1.dylib, etc. but the file is not found in any of the locations. I tried to reinstall libvoikko but I am still facing same problem. Homebrew says that it should support Apple silicon/Venture. Any ideas?

**Comment by hatapitk** [comment]
Hi! Can you find the location where the missing library is installed? I do not have Silicon hardware but I do have an older Mac and could take a look some day. Perhaps we need some changes in the Python module to handle this.

**Comment by f4irline** [comment]
I had this problem as well with the Java bindings. Homebrew installs the library file to `/opt/homebrew/Cellar/libvoikko/4.3.2/lib/libvoikko.1.dylib`. You might have to copy it to a directory which Voikko checks, so probably e.g. `/usr/lib/libvoikko.1.dylib`?

**Reply by jarirajari** [reply]
Yes, I also finally ended up doing a workaround like this i.e. manually copying the missing library file to a location where it gets picked up.

**Comment by hatapitk** [comment]
I tested this on freshly installed Intel MacOS 13.4 and had no problems when `brew --prefix` is /usr/local. If it is something else there could be problems (see for example https://github.com/Homebrew/brew/issues/9214 ). I'm not sure if there is anything reasonable I could do in the code to fix this. In theory I could fall back to calling `brew --prefix` if the library is not found but that could be considered a bit unexpected by some.

Instead of copying the library you could also call `libvoikko.Voikko.setLibrarySearchPath("/opt/homebrew/lib")` before trying to create your Voikko instance. Maybe that helps?

**Comment by jarirajari** [comment]
I think this discussion is over as there are solutions to the problem. Thank you for helping me!
